includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1117,7 +1117,25 @@ public function process_payment_for_order( $cart, $payment_information, $schedul
 				$request->set_off_session( $payment_information->is_merchant_initiated() );
 				$request->set_payment_methods( $payment_methods );
 				$request->set_cvc_confirmation( $payment_information->get_cvc_confirmation() );
-				// Make sure that setting fingerprint is performed after setting metadata becaouse metadata will override any values you set before for metadata param.
+
+				if ( 1 === count( $payment_methods ) && 'card' !== $payment_methods[0] ) {
+					$request->set_return_url(
```

Comment:
1. This is a good callout! SEPA support was not considered part of this PR, which was basically because it's not fully compatible with the split UPE and is currently disabled. Attaching the `return_url` to this payment method doesn't make sense, but currently, SEPA won't work even without attaching the `return_url`. I'll consult the creation of an issue to implement SEPA for the deferred intent creation with the team and will link it here.
2. Fair suggestion - thank you! I've updated the code in 42ac641c74405b95b19fca442e56813e30d7bb46